object-construction-checker/tests/socket/InputOutputStreams.java

```diff
@@ -0,0 +1,100 @@
+// A MustCallChoice test wrt sockets. Also coincidentally tests that MCC sets can be larger than two.
+
+import java.io.*;
+import java.net.*;
+import org.checkerframework.checker.objectconstruction.qual.*;
+import java.io.IOException;
+
+class InputOutputStreams {
+    void test_close_sock(@Owning Socket sock) throws IOException {
+        try{
+            InputStream is = sock.getInputStream();
+            OutputStream os = sock.getOutputStream();
+        } catch (IOException e) {
+
+        } finally {
+            sock.close();
+        }
+    }
+
+    // :: error: required.method.not.called
+    void test_close_is(@Owning Socket sock) throws IOException {
```

Comment:
The reason that this test (and several of the following) should fail is subtle, because on first glance they look correct. Can we add a comment about how `getInputStream` might throw an exception, making `is` null in the finally block - thus causing the socket not to be closed? I think we only need that note once - it also explains why e.g. `test_close_os` below should result in a warning. I'm asking because I screwed that up when I originally wrote these tests :)